server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -3950,8 +3950,8 @@ ReplicationTracker getReplicationTracker() {
     /**
      * Executes a scheduled refresh if necessary. Completes the listener with true if a refresh was performed otherwise false.
      */
-    public void scheduledRefresh(ActionListener<Boolean> listener) {
-        ActionListener.run(listener, l -> {
+    public boolean scheduledRefresh(ActionListener<Boolean> listener) {
```

Comment:
I don't think it's necessary to know we sent the refresh request or not. See my other comment, I believe the assertBusy solution was sufficient.

Reply:
Hey @kingherc! I liked the `assertBusy` option better, too, because we don't need to go the implementation details of `scheduledRefresh`. I also found that `SearchIdleIT` faced a a similar issue and the test worked around it by blocking the refresh pool which also seems quite hacky.

I've reverted my last commit and kept the `assertBusy` solution. It seems that it indeed should be reliable due the addition of the second document.

Reply:
(I don't see the assertBusy yet.)

Reply:
Sorry, the commit didn't get pushed due to network issues. Should be visible now.